packages/examples/packages/ethereum-provider/src/index.ts

```diff
@@ -207,6 +223,9 @@ async function signTypedData(message: string, from: string) {
  * @see https://docs.metamask.io/snaps/reference/rpc-api/#wallet_invokesnap
  */
 export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
+  const { chainId = '0x1' } = (request.params as BaseParams) ?? {};
```

Comment:
i don't know about about snaps, but I'm assuming that this type cast has to happen here and can't be moved into the function params type